lib/SIL/SILFunctionType.cpp

```diff
@@ -1035,18 +1216,18 @@ static CanSILFunctionType getSILFunctionType(
                                         substFormalResultType);
   }
 
-  // Destructure the result tuple type.
-  SmallVector<SILResultInfo, 8> results;
-  {
-    DestructureResults destructurer(expansionContext, TC, conventions, results);
-    destructurer.destructure(origResultType, substFormalResultType);
-  }
+  SubstFunctionTypeCollector subst(TC,
+    TC.Context.LangOpts.EnableSubstSILFunctionTypesForFunctionValues
+    // We don't currently use substituted function types for generic function
+    // type lowering, though we should for generic methods on classes and
+    // protocols.
+    && !genericSig);
```

Comment:
`subst` seems like a pretty generic name for this variable; maybe `substCollector`, or just anything that suggests it's not a substituted type of some sort?